src/data/feed/quandl.js

```diff
@@ -0,0 +1,146 @@
+import d3 from 'd3';
+
+//  https://www.quandl.com/docs/api#datasets
+export default function() {
+
+    var _columnNameMap = function(colName) {
+        colName = colName.replace(/[\s()-.]/g, '');
+        return colName.substr(0, 1).toLowerCase() + colName.substr(1);
+    };
+
+    var database = 'YAHOO',
+        dataset = 'GOOG',
+        apiKey = null,
+        start = null,
+        end = null,
+        rows = null,
+        descending = false,
+        collapse = null,
+        columnNameMap = _columnNameMap;
+
+    var quandl = function(cb) {
+        var params = [];
+        if (apiKey != null) {
+            params.push('api_key=' + apiKey);
+        }
+        if (start != null) {
+            params.push('start_date=' + start.toISOString().substring(0, 10));
+        }
+        if (end != null) {
+            params.push('end_date=' + end.toISOString().substring(0, 10));
+        }
+        if (rows != null) {
+            params.push('rows=' + rows);
+        }
+        if (!descending) {
+            params.push('order=asc');
+        }
+        if (collapse != null) {
+            params.push('collapse=' + collapse);
+        }
+
+        var url = 'https://www.quandl.com/api/v3/datasets';
+        url += ('/' + database + '/' + dataset + '/').replace(/\/\//g, '');
+        url += 'data.json?' + params.join('&');
+
+        d3.json(url, function(error, data) {
+            if (error) {
+                cb(error);
+                return;
+            }
+
+            data = data.dataset_data;
+            var colNames = data.column_names.map(columnNameMap);
+
+            data = data.data.map(function(d) {
+                var output = {};
+
+                output[colNames[0]] = new Date(d[0]);
+                for (var i = 1; i < colNames.length; i++) {
+                    if (colNames[i] != null) {
```

Comment:
How about using `Array.filter` when you create `colNames` removing the need for this repeated check.